### ISSUE TITLE
Loss value 'nan'

### ISSUE THREAD

**Issue by nargesdel** [neulab/guided_summarization#53]
Hello,

I am training the bart model for 5 epochs. While training the model, I have a defined value for loss in the first epoch. However, the value of loss for the rest of epochs is 'nan'. I am wondering if there is some problems with my training or it is normal.
It is mentioned:

epoch 002:  18%|▏| 1386/7494 [06:49<29:04,  3.50it/s, loss=nan, nll_loss=nan, ppl=nan, wps=216.1, ups=3.4, wpb=63.5, bsz=1.1, num_updates=8880, lr=1.71077e-05, gnorm=na
epoch 002:  19%|▏| 1387/7494 [06:49<28:22,  3.59it/s, loss=nan, nll_loss=nan, ppl=nan, wps=216, ups=3.4, wpb=63.5, bsz=1.1, num_updates=8881, lr=1.71062e-05, gnorm=nan,

Here is my z_train.sh:

#!/bin/sh
TOTAL_NUM_UPDATES=20000  
WARMUP_UPDATES=500      
LR=3e-05
MAX_TOKENS=512
UPDATE_FREQ=1
BART_PATH=/home/nlp/bart.large/model.pt
DATA_BIN=/home/nlp/neulab/guided_summarization/data/cnn_dm/bin
SAVE_DIR=/home/nlp/neulab/guided_summarization/data/model_result

CUDA_VISIBLE_DEVICES=0,1 python train.py $DATA_BIN \
    --restore-file $BART_PATH \
    --max-tokens $MAX_TOKENS \
    --max-epoch 5 \
    --task guided_translation \
    --source-lang source --target-lang target \
    --truncate-source \
    --layernorm-embedding \
    --share-all-embeddings \
    --share-decoder-input-output-embed \
    --reset-optimizer --reset-dataloader --reset-meters \
    --required-batch-size-multiple 1 \
    --arch guided_bart_large \
    --criterion guided_label_smoothed_cross_entropy \
    --label-smoothing 0.1 \
    --dropout 0.1 --attention-dropout 0.1 \
    --weight-decay 0.01 --optimizer adam --adam-betas "(0.9, 0.999)" --adam-eps 1e-08 \
    --clip-norm 0.1 \
    --lr-scheduler polynomial_decay --lr $LR --total-num-update $TOTAL_NUM_UPDATES --warmup-updates $WARMUP_UPDATES \
    --update-freq $UPDATE_FREQ \
    --skip-invalid-size-inputs-valid-test \
    --save-dir $SAVE_DIR \
    --find-unused-parameters;

Thank you very much for your consideration.

**Comment by nargesdel** [comment]
Does anybody has an idea about this?
Thanks in advance.

**Comment by zdou0830** [comment]
Hi, this is not normal. I am not sure how to solve this but I noticed that there are some differences between your parameter settings and [this](https://github.com/neulab/guided_summarization/blob/master/bart/z_train.sh). could you try to follow the settings in https://github.com/neulab/guided_summarization/blob/master/bart/z_train.sh?

**Comment by nargesdel** [comment]
Hello,
Thank you very much for your reply. I needed to use my own setting regarding my GPU machine.
I have changed my guidance signal data and the problem was resolved. I believe there was a problem in that file.
Thank you for your consideration.
